templates/gitlab-manager-runner.yaml

```diff
@@ -0,0 +1,542 @@
+              - !Sub 'arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*'
+            Condition:
+              StringLike:
+                'ec2:ResourceTag/Name': '*gitlab-docker-machine-*'
+              ArnEquals:
+                'ec2:InstanceProfile': !GetAtt 'RunnersInstanceProfile.Arn'
+          - Effect: 'Allow'
+            Action:
+              - 'iam:PassRole'
+            Resource:
+              - !GetAtt 'RunnersRole.Arn'
+  AccessSecretPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+              - kms:Decrypt
+            Resource:
+              - !Ref 'KmsKeyArn'
+              - !Ref 'DbCredsArn'
+          - Effect: Allow
+            Action:
+              - logs:CreateLogStream
+            Resource: '*'
+
+  ####################################
+  ### S3 Bucket for Runners' cache ###
+  ####################################
+  CacheBucket:
+    Type: 'AWS::S3::Bucket'
+    Properties:
+      LifecycleConfiguration:
+        Rules:
+          - Status: 'Enabled'
+            ExpirationInDays: 30
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+      Tags:
+        - Key: Name
+          Value: !Sub '${AWS::StackName}-${AWS::AccountId}-CiRunner'
+
+  ##############
+  ### Groups ###
+  ##############
+  ManagerSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      VpcId: !Ref 'VpcId'
+      GroupDescription: 'Security group for the GitLab Runner Manager.'
+      SecurityGroupIngress:
+        - Description: 'SSH traffic'
+          IpProtocol: 'tcp'
+          FromPort: 22
+          ToPort: 22
+          CidrIp: '0.0.0.0/0'
+      Tags:
+        - Key: Name
+          Value: !Sub '${AWS::StackName}-CiRunner-Manager'
+  RunnerAutoScalingGroup:
+    Type: 'AWS::AutoScaling::AutoScalingGroup'
+    Properties:
+      LaunchConfigurationName: !Ref Manager
+      VPCZoneIdentifier: !Ref PrivateSubnetIds
+      MinSize: !Ref MinNumInstances
+      MaxSize: !Ref MaxNumInstances
+      Cooldown: '300'
+      DesiredCapacity: !Ref MinNumInstances
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName}
+          PropagateAtLaunch: true
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref MinNumInstances
+        Timeout: PT15M
+
+  #############
+  ### Roles ###
+  #############
+  ManagerRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'ec2.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM
+        - !Ref CachePolicy
```

Comment:
Need to add `arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore` if this is the policy list that ends up in the EC2 profile for the gitlab runner. We use SSM to manage EC2s. The instances need to have the tag `PatchGroup` on them with the value `prod-default`